lib/Slackbot_worker.js

```diff
@@ -86,10 +105,22 @@ module.exports = function(botkit, config) {
             bot.rtm = new Ws(res.url, null, {agent: agent});
             bot.msgcount = 1;
 
-            var pingIntervalId = null;
+            bot.rtm.on('pong', function (obj) {
+                lastPong = Date.now()
+                botkit.debug('PONG received');
+            })
+
             bot.rtm.on('open', function() {
 
                 pingIntervalId = setInterval(function() {
+                    if (lastPong && lastPong + 12000 < Date.now()) {
```

Comment:
Hmm. If you are interactively debugging your bot and stepping through for a bit you will miss the pong and then get this Stale RTM killing the connection. Any thoughts on whether bypassing the pong if in debug mode is a valid idea?

// http://stackoverflow.com/a/13454643
var debug = typeof v8debug === 'object';